drivers/overlay/overlay.go

```diff
@@ -2162,11 +2162,14 @@ func supportsDataOnlyLayersCached(home, runhome string) (bool, error) {
 // ApplyDiffWithDiffer applies the changes in the new layer using the specified function
 func (d *Driver) ApplyDiffWithDiffer(id, parent string, options *graphdriver.ApplyDiffWithDifferOpts, differ graphdriver.Differ) (output graphdriver.DriverWithDifferOutput, errRet error) {
 	var idMappings *idtools.IDMappings
+	forceMask := d.options.forceMask
+
 	if options != nil {
 		idMappings = options.Mappings
+		forceMask = options.ForceMask
 	}
 	if idMappings == nil {
-		idMappings = &idtools.IDMappings{}
+		idMappings = idtools.NewIDMappingsFromMaps(d.uidMaps, d.gidMaps)
```

Comment:
Warning: I don’t understand the whole picture at all.

Shouldn’t this happen at a (much?) higher level? Looking at `store.putLayer`, it is done there — and `canUseShifting` is involved in the decision.

I have no idea which of the two is correct, to be clear.

Reply:
I never understood this part as well :-)  I took the chance now to dig deeper into it...

I think we are carrying the feature from Docker (e5f58bf9c25af5d7a49adbe1b597bcd1e3f33772) where it was possible to configure a single uid/gid mapping for all the containers, so the mappings are configured only once for the driver.

In facts this is the setting we get when we configure `remap-uids` and `remap-gids` in the storage.conf file.

I'll drop this patch from the current PR and remove the `remap-*ids` logic completely in a different PR, as it has been broken forever

Reply:
opened a test PR to see how far the CI goes: https://github.com/containers/storage/pull/1976

Reply:
If I read the code correctly, the `d.[ug]idMaps` fields are never used when applying tarballs; and the generic code which triggers shifting or per-container remapping does not know about this option; that makes it really hard to see a use case.

Dealing with that in a separate PR LGTM.